src/NServiceBus.Metrics/MetricsFeature.cs

```diff
@@ -12,11 +14,26 @@
     /// </summary>
     public class MetricsFeature : Feature
     {
+        /// <summary>
+        /// 
+        /// </summary>
+        public MetricsFeature()
+        {
+            Prerequisite(c => !c.Settings.Get<bool>("Endpoint.SendOnly"), "Processing metrics are not supported on send-only endpoints");
+            Defaults(settings =>
+            {
+                var bindingFlags = BindingFlags.NonPublic | BindingFlags.Instance;
+                // Unfortunately the constructor is internal, so we have to use reflection to create an instance of RecoverabilitySettings
+                var recoverability = (RecoverabilitySettings)Activator.CreateInstance(typeof(RecoverabilitySettings), bindingFlags, null, [settings], null);
```

Comment:
Does this newly created `RecoverabilitySettings` have to somehow be connected to the `EndpointConfiguration` for it to take effect?

Reply:
That is done by passing the settingsHolder bag into the constructor.